### PR TITLE
Move configuration-based client logic to Microsoft.Azure.WebJobs.Extensions.Clients

### PR DESCRIPTION
**Pull request by pakrym**
Considering the idea that `Microsoft.Extensions.Azure` would be able to handle all logic of client creation by only knowing the connection name and having IConfiguration instance failed and we moved to it providing helpers to create client "components" via `AzureComponentFactory` I'm removing the feature from `Microsoft.Extensions.Azure`

### CODE REVIEW

File: sdk/storage/Azure.Storage.Webjobs.Extensions.Blobs/src/Azure.WebJobs.Extensions.Storage.Blobs.csproj [hunk 1]
```diff
@@ -13,8 +13,9 @@
 
   <ItemGroup>
     <Compile Include="$(AzureStorageWebjobsExtensionSharedSources)\**\*.cs" Link="Shared\%(RecursiveDir)\%(Filename)%(Extension)" />
+    <Compile Include="..\..\..\extensions\Microsoft.Azure.WebJobs.Extensions.Clients\src\Shared\*.cs" Link="Shared\%(RecursiveDir)\%(Filename)%(Extension)" />
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Webjobs.Extensions.Blobs/src/Azure.WebJobs.Extensions.Storage.Blobs.csproj:16]: could you create a "variable" to capture this path in one place and use in both blobs and queues extension?
also, would that "Shared" get any internal structure? might be worth using `/**/`

File: sdk/storage/Azure.Storage.Webjobs.Extensions.Common/src/Shared/Constants.cs [hunk 2]
```diff
@@ -5,11 +5,6 @@ namespace Microsoft.Azure.WebJobs.Extensions.Storage.Common
 {
     internal static class Constants
     {
-        public const string WebJobsConfigurationSectionName = "AzureWebJobs";
-        public const string EnvironmentSettingName = "AzureWebJobsEnv";
-        public const string DevelopmentEnvironmentValue = "Development";
-        public const string DynamicSku = "Dynamic";
-        public const string AzureWebsiteSku = "WEBSITE_SKU";
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Webjobs.Extensions.Common/src/Shared/Constants.cs:12]: where did this go?

File: sdk/storage/Azure.Storage.Webjobs.Extensions.Queues/src/StorageQueuesWebJobsBuilderExtensions.cs [hunk 3]
```diff
@@ -33,7 +33,7 @@ public static class StorageQueuesWebJobsBuilderExtensions
         /// <returns></returns>
         public static IWebJobsBuilder AddAzureStorageQueues(this IWebJobsBuilder builder, Action<QueuesOptions> configureQueues = null)
         {
-            builder.Services.AddAzureClients(_ => { });
+            builder.Services.AddAzureClientsCore();
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Webjobs.Extensions.Queues/src/StorageQueuesWebJobsBuilderExtensions.cs:36]: I guess similar change is due in blobs part.

**Review by kasobol-msft — COMMENTED** [review]
Overall looks good to me.

File: sdk/storage/Azure.Storage.Webjobs.Extensions.Common/src/Shared/Constants.cs (hunk 2 above)
Comment (pakrym) [reply sdk/storage/Azure.Storage.Webjobs.Extensions.Common/src/Shared/Constants.cs:12]: Wasn't used

Comment (kasobol-msft) [reply sdk/storage/Azure.Storage.Webjobs.Extensions.Common/src/Shared/Constants.cs:12]: Ah, these were copy pasted elsewhere at some point.

**Review by kasobol-msft — APPROVED** [review]
LGTM.

**Comment by pakrym** [comment]
/

**Pull request merged**